src/logic/wallets/store/actions/fetchProvider.js

```diff
@@ -44,10 +44,24 @@ const handleProviderNotification = (openSnackbar: Function, provider: ProviderPr
   openSnackbar(msg, variant)
 }
 
+export const recurrentFetchProvider = (currentProvider: ProviderProps, openSnackbar: Function) => async (
```

Comment:
why do we need this action? 
Imo the `fetchProvider`  action we currently have can be refactored to this